engine/src/main/java/org/terasology/engine/core/module/ModuleManager.java

```diff
@@ -73,17 +73,32 @@ public ModuleManager(String masterServerAddress, List<Class<?>> classesOnClasspa
 
         registry = new TableModuleRegistry();
 
-        engineModule = loadEngineModule(classesOnClasspathsToAddToEngine);
+        engineModule = moduleFactory.createPackageModule("org.terasology.engine");
+        Module nui = createSyntheticPackageModule("nui", "3.0.0", "org.terasology.nui");
+        Module discordSubsystem = moduleFactory.createPackageModule("org.terasology.subsystem.discordrpc");
 
-        loadModulesFromApplicationPath(pathManager);
+        createDependency(engineModule, nui);
+
+        registry.add(nui);
+        registry.add(engineModule);
+        registry.add(discordSubsystem);
 
+        loadModulesFromApplicationPath(pathManager);
         ensureModulesDependOnEngine();
 
         setupSandbox();
-        loadEnvironment(Sets.newHashSet(engineModule), true);
+        loadEnvironment(Sets.newHashSet(engineModule, nui, discordSubsystem), true);
```

Comment:
TODO: We should switch this to the new `resolveAndLoadEnvironment` method. It'll do dependency resolution so we don't have to manually keep these lists in sync with every change.